### PR TITLE
Update javadoc

### PR DESCRIPTION
**Pull request by Simson475**
Added comments.
Cleanup was done
Inheritance hierarchy was made for documents.
Faulty test commented out.

### CODE REVIEW

File: src/main/java/searchengine/booleanretrieval/DataSelection.java [hunk 1]
```diff
@@ -1,27 +1,35 @@
...
 import java.sql.Connection;
 import java.sql.DriverManager;
 import java.sql.ResultSet;
 import java.sql.Statement;
 import java.util.ArrayList;
 import java.util.List;
+import searchengine.vsm.TFIDFDocument;
 
+/**
+ * Contains the retrieveDocuments class.
```
Comment (BjoerklundJacob) [inline src/main/java/searchengine/booleanretrieval/DataSelection.java]:
```suggestion
 * Contains the retrieveDocuments method.
```

File: src/main/java/searchengine/booleanretrieval/DataSelection.java [hunk 2]
```diff
@@ -1,27 +1,35 @@
...
+ * Is used for retrieving document names.
+ */
 public class DataSelection {
 
-    /*
-    *@param input: the search query in the ofrmat "term1 term2 term3 term3" (space separated)
-    * @return a list of documents which contained at least one of the search terms
+    /**
+     * Searches the wordcount database for documents containing the terms from the input string.
+     * @param input: The search query in the format "term1 term2 term3 term3" (space separated)
```
Comment (BjoerklundJacob) [inline src/main/java/searchengine/booleanretrieval/DataSelection.java:19]:
```suggestion
     * @param input: The search query in the format "term1 term2 ... termN" (space separated)
```

File: src/main/java/searchengine/booleanretrieval/DataSelection.java [hunk 3]
```diff
@@ -53,13 +61,12 @@
         return documents;
     }
 
-    /*
-    * @param input: the search query in the ofrmat "term1 term2 term3 term3" (space separated)
-    * @return the SQL query
+    /** SubFunction in retrieveDocuments. Generates an SQL query that searches for the terms from the input string.
+     * The input string should be split with space between terms
+     * @param input: the search query in the ofrmat "term1 term2 term3 term3" (space separated)
```
Comment (BjoerklundJacob) [inline src/main/java/searchengine/booleanretrieval/DataSelection.java:66]:
```suggestion
     * @param input: the search query in the ofrmat "term1 term2 ... termN" (space separated)
```

File: src/main/java/searchengine/restcontrollers/Search.java [hunk 4]
```diff
@@ -2,39 +2,36 @@
...
 import java.net.URL;
-import java.net.http.HttpClient;
-import java.net.http.HttpRequest;
-import java.net.http.HttpResponse;
 import java.util.List;
-import java.util.Map;
 
 
+/**
+ * Encapsulates a search for some documents. the resulting documents are ranked and stored in the result property.
```
Comment (BjoerklundJacob) [inline src/main/java/searchengine/restcontrollers/Search.java:18]:
```suggestion
 * Encapsulates a search for some documents. The resulting documents are ranked and stored in the result property.
```

File: src/main/java/searchengine/restcontrollers/Search.java [hunk 5]
```diff
@@ -43,7 +40,8 @@ public Search(String input) throws IOException, JSONException {
     }
 
 
-    /*
+    /**
+     * lemmatizes the string from the input. should always be called before sending a query to the wordcount database
```
Comment (BjoerklundJacob) [inline src/main/java/searchengine/restcontrollers/Search.java:44]:
```suggestion
     * Lemmatizes the string from the input. Should always be called before sending a query to the wordcount database
```

File: src/main/java/searchengine/vsm/VectorSpaceModel.java [hunk 6]
```diff
@@ -2,129 +2,152 @@
...
-            }
-        }
+        List<ScoredDocument> scoredDocuments = new ArrayList<>();
+        calculateTFIDF(documents);
 
-        // Get the inverse document frequency of each term
-        for (String term : df.keySet()) {
-            idf.put(term, Math.log10(documents.size() / (double)df.get(term)));
+        for (TFIDFDocument doc : documents) {
+            Double score = cosineSimilarityScore(doc);
```
Comment (BjoerklundJacob) [inline src/main/java/searchengine/vsm/VectorSpaceModel.java:54]:
```suggestion
            double score = cosineSimilarityScore(doc);
```

File: src/test/java/searchengine/vsm/VectorSpaceModelTest.java [hunk 7]
```diff
@@ -10,51 +10,51 @@
...
     static VectorSpaceModel vsm;
-    static List<Document> documents;
+    static List<TFIDFDocument> TFIDFDocuments;
     static List<ScoredDocument> scoredDocuments;
 
     @BeforeAll
     static void setup() {
         // arrange
-        documents = new ArrayList<>();
+        TFIDFDocuments = new ArrayList<>();
```
Comment (BjoerklundJacob) [inline src/test/java/searchengine/vsm/VectorSpaceModelTest.java:24]: Måske rename til tfidfDocuments

**Review by Zzeeker — APPROVED** [review]

**Pull request merged**